app/src/main/kotlin/com/wire/android/mapper/MessagePreviewContentMapper.kt

```diff
@@ -38,61 +38,61 @@ fun MessagePreview?.toUIPreview(unreadEventCount: UnreadEventCount): UILastMessa
-                        type.value,
-                        type.value
-                    )
-
-                    UnreadEventType.MISSED_CALL -> UnreadEventType.MISSED_CALL to UIText.PluralResource(
-                        R.plurals.unread_event_call,
-                        type.value,
-                        type.value
-                    )
-
-                    UnreadEventType.MENTION -> UnreadEventType.MENTION to UIText.PluralResource(
-                        R.plurals.unread_event_mention,
-                        type.value,
-                        type.value
-                    )
-
-                    UnreadEventType.REPLY -> UnreadEventType.REPLY to UIText.PluralResource(
-                        R.plurals.unread_event_reply,
-                        type.value,
-                        type.value
-                    )
-
-                    UnreadEventType.MESSAGE -> UnreadEventType.MESSAGE to UIText.PluralResource(
-                        R.plurals.unread_event_message,
-                        type.value,
-                        type.value
-                    )
-
-                    UnreadEventType.IGNORED -> null
-                    null -> null
-                }
-            }.associate { it }
-        if (unreadContentTexts.size > 1) {
-            val first = unreadContentTexts.values.first()
-            val second = unreadContentTexts.values.elementAt(1)
-            return UILastMessageContent.MultipleMessage(listOf(first, second))
-        } else if (unreadContentTexts.isNotEmpty()) {
-            val unreadContent = unreadContentTexts.entries.first()
-            if (unreadContent.key != UnreadEventType.MESSAGE) {
-                return UILastMessageContent.TextMessage(MessageBody(unreadContent.value))
+                UnreadEventType.MISSED_CALL -> UnreadEventType.MISSED_CALL to UIText.PluralResource(
+                    R.plurals.unread_event_call,
+                    type.value,
+                    type.value
+                )
+
+                UnreadEventType.MENTION -> UnreadEventType.MENTION to UIText.PluralResource(
+                    R.plurals.unread_event_mention,
+                    type.value,
+                    type.value
+                )
+
+                UnreadEventType.REPLY -> UnreadEventType.REPLY to UIText.PluralResource(
+                    R.plurals.unread_event_reply,
+                    type.value,
+                    type.value
+                )
+
+                UnreadEventType.MESSAGE -> UnreadEventType.MESSAGE to UIText.PluralResource(
+                    R.plurals.unread_event_message,
+                    type.value,
+                    type.value
+                )
+
+                UnreadEventType.IGNORED -> null
+                null -> null
             }
-        }
-    }
+        }.associate { it }
 
-    return uiLastMessageContent()
+    val first = unreadContentTexts.values.first()
+    val second = unreadContentTexts.values.elementAt(1)
```

Comment:
I believe this method will be called also when `unreadEventCount.size == 1 && unreadEventCount.keys.first() != UnreadEventType.MESSAGE && unreadEventCount.values.first() != 1`, in that case there won't be any element at position 1

Reply:
Good catch 👨‍🏫